cooldowns/trigger_cooldown.py

```diff
@@ -0,0 +1,266 @@
+            ```
+
+                From the moment when the cooldown was triggered by `.trigger(30)`, every
+                single call to this command within 30 seconds will raise CallableOnCooldown!
+        
+        Raises
+        ------
+        `RuntimeError`
+            Expected the decorated function to be a coroutine.
+        `CallableOnCooldown`
+            This call resulted in a cooldown being put into effect.
+        """
+
+        self.limit = limit
+        self.time_period = time_period
+        self.bucket = bucket
+        self.cooldown_id = cooldown_id
+        self.trigger_cooldown_id = trigger_cooldown_id
+        self.check = check
+
+        # Normal Cooldown
+        self.cooldown = Cooldown(
+                limit= self.limit,
+                time_period= self.time_period,
+                bucket= self.bucket,
+                cooldown_id= self.cooldown_id,
+                check= self.check
+            )
+
+        # Trigger Cooldown
+        self.trigger_cooldown = Cooldown(
+                limit= 1,
+                time_period= self.time_period,
+                bucket= self.bucket,
+                cooldown_id= self.trigger_cooldown_id,
+                check= self.check
+            )
+
+        if cooldown_id:
+            utils.shared_cooldown_refs[cooldown_id] = self.cooldown
+
+        else:
+            current_cooldowns = utils.shared_cooldown_refs.keys()
+            for i in range(10_000):
+                generated_id = f"normal_cooldown_{i:02}"
+                if generated_id not in current_cooldowns:
+                    utils.shared_cooldown_refs[generated_id] = self.cooldown
+                    self.cooldown_id = generated_id
+
+        if trigger_cooldown_id:
+            utils.shared_cooldown_refs[trigger_cooldown_id] = self.trigger_cooldown
+
+        else:
+            current_cooldowns = utils.shared_cooldown_refs.keys()
+            for i in range(10_000):
+                generated_id = f"trigger_cooldown_{i:02}"
+                if generated_id not in current_cooldowns:
+                    utils.shared_cooldown_refs[generated_id] = self.trigger_cooldown
+                    self.trigger_cooldown_id = generated_id
+
+    async def trigger(self, time_period: Union[float, datetime.timedelta]) -> None:
+        """|coro|
+
+        Trigger the Trigger Cooldown instantly. Has to be awaited.
+
+        Parameters
+        ----------
+        time_period : `Union[float, datetime.timedelta]`
+            The time period that cooldwon will remain triggered.
+        """
+        self.trigger_cooldown.time_period = (
+            time_period
+            if isinstance(time_period, (float, int))
+            else time_period.total_seconds()
+        )
+
+        # Triggers the Cooldown leaving bucket.current = 0
+        frame = inspect.currentframe().f_back
+        _, _, _, values = inspect.getargvalues(frame)
+        args = tuple(values.values())
```

Comment:
I disagree with implicit argument fetching here, I'd prefer if an end user provided the arguments to `trigger` for a few reasons:
- This doesn't handle nested situations or situations where trigger is called outside the context of the command it decorates
- This doesn't ideally handle buckets, I.e. if you want to trigger for a specific bucket

If this trigger is meant to affect every call regardless, why does it pass args?